packages/scripts/config/webpack.config.js

```diff
@@ -36,9 +37,8 @@ if ( ! browserslist.findConfig( '.' ) ) {
 }
 const hasReactFastRefresh = hasArgInCLI( '--hot' ) && ! isProduction;
 
-const copyWebpackPatterns = process.env.WP_COPY_PHP_FILES_TO_DIST
-	? '**/{block.json,*.php}'
-	: '**/block.json';
+// Get paths of the `render` props included in `block.json` files
+const renderPaths = getRenderPropPaths();
```

Comment:
It would be better to call this function in the filter, so changes applied to `block.json` files are applied correctly in the watch mode.

Reply:
Taking the function out of the filter was actually my suggestion to avoid reading/parsing all the `block.json` files per each PHP file of the project. That can slow things down for projects with multiple blocks and hundreds of PHP files (imagine something similar to WooCommerce).

Webpack is not watching changes in either JSON or PHP files anyway, or is it?

Reply:
Ah, I see what you mean here. It would have to run the same function for every PHP file whenever watch mode triggers. It's different from how we calculate entry points which runs only once per rebuild:

https://github.com/WordPress/gutenberg/blob/e7b26f64d1fe369aa47a956ff0afa8b269d735ba/packages/scripts/config/webpack.config.js#L93

> Webpack is not watching changes in either JSON or PHP files anyway, or is it?

It's a different issue. This code is inside webpack config that is executed only once when you run `wp-scripts start`. That's why we would be in better position if we could use a callback for the `CopyWebpackPlugin` so it detects changes to the `block.json` file. We can always leave it as is if we don't come up with an efficient caching strategy.

Reply:
> Webpack is not watching changes in either JSON or PHP files anyway, or is it?

Oh, the ones added to the copy plugin are 🙂

> We can always leave it as is if we don't come up with an efficient caching strategy.

Sounds good to me. I can't think of a suitable caching strategy that would be simpler to implement right now.

Reply:
> > Webpack is not watching changes in either JSON or PHP files anyway, or is it?
>
> Oh, the ones added to the copy plugin are 🙂

Nice, so it should work in most of the cases. The only exception would be when a new `block.json` is added with `render`, or when the existing `block.json` gets any modifications to the `render` field.

Reply:
I think that's correct. @SantosGuillamot can you confirm that?

Reply:
From the test I did, that's right. Modifications to the `render.php` file seem to be applied, but not the changes made to the `block.json`.

Reply:
Excellent, thank you for the confirmation.

Reply:
I have just realized that we previously discussed the issue filed under https://github.com/WordPress/gutenberg/issues/49790. There is a fix proposed in https://github.com/WordPress/gutenberg/pull/50939, but it goes against the reasoning covered in the discussion above. What do you recommend as the next step?

Reply:
I opened a PR https://github.com/WordPress/gutenberg/pull/51162 where I propose a webpack plugin that ensures that `renderPaths` updates only once per update detected in the watch mode.